\section{Introduction}

As the availability of genetic data has increased,
so has our ability to infer evolutionary processes at finer scales.
An essential tool for evolutionary inference has been simulations,
and many flexible and efficient evolutionary simulators have been developed over the past decade \citep{hudson_testing_1983, hudson_test_1987, carvajal-rodriguez_simulation_2008, ohta_simulation_1974, hoban_computer_2012, bulmer_effect_1976}.
Until recently, little had been done to integrate different tools and to ensure compatibility of inferred models across studies.

A unifying thread that has emerged is the tree sequence data structure \citep{kelleher_efficient_2016}.
The tree sequence provides a way of concisely representing correlated genealogies along chromosomes.
Both msprime, an efficient coalescent simulator, and SLiM, the most widely used forward-in-time simulator,
record the genealogical history of all samples in a population in the form of tree sequences \citep{kelleher_efficient_2018, haller_tree-sequence_2019, haller_slim_2019}.
One of the major advantages of doing so in forward-in-time simulations is that it allows neutral mutations to be omitted.
By definition, such mutations do not impact the underlying genealogies, but they pose immense computational demand (due to bookkeeping).
Therefore, by omitting the neutral mutations from the forward step, it is possible to decrease computational resources dramatically.
After completion of the forward simulation, it is possible to simply overlay the genealogies with the neutral mutations if necessary.

As inference becomes ever more complex, now enabled by powerful simulation tools,
the need for easy and error-free dissemination of estimated models has increased.
Many researchers used to re-implement models independently,
a process which is error-prone and results in duplication of effort.
Indeed, a recent study described errors in the implementation of a demographic model in two published papers,
which affected the interpretation of biological signals \citep{ragsdale_lessons_2020}.

Stdpopsim is a community-driven open source package developed to minimize these issues \citep{adrion_community-maintained_2020, lauterbur_expanding_2023}.
We organized a well-documented library with published simulation models from a range of organisms,
which can be accessed with simple Python and command-line interfaces.
The library contains demographic models, recombination maps and mutation rates for tens of species.
All the models have to pass a quality control method, in which an independent party validates the model.
Further, we provide a Python API to easily run simulations using the catalog, with msprime and SLiM as simulators on the backend,
decreasing even further the barrier to reproducing simulations.

In this chapter, my contributions to the current ecosystem of evolutionary simulation tools in two vignettes.
First, I describe a method I devised to make multi-population simulations faster with parallelization, which is enabled by tree sequences.
Second, I report my main contribution to \stdpopsim: the inclusion of selection models.
Using this new feature, I investigate how the power to detect sweeps along more realistic chromosomes.
Both of these contributions will appear (in some form) in future publications of the tree sequence toolkit library (\tskit) and of the \stdpopsim consortium.


\section{A way to parallelize multi-population simulations with tree sequences}

A major problem in simulation-based inference is the computational cost of simulations.
In many cases, it is possible to minimize the time cost of simulations by running them in parallel,
that is to divide up the simulation over many processes that can be executed concurrently over many CPUs (or cores).
It is not always clear how to divide a simulation into sub-tasks, however.

In the case of evolutionary simulations, a natural way to break up a big simulation might be population splits.
After a population splits into two (or more) subpopulations,
their histories become independent (assuming no migration).
Therefore, it is possible to parallelize a multi-population simulation by dividing the simulation over population splits.
The history of populations A, B and C (shown in \lcref{fig:pop_hist}) can be parallelized:
any two branches stemming from the same node are independent.

\begin{figure}[htp]
\includegraphics[width=0.5\linewidth]{{union_example/phylo.png}}
\centering
\caption[Example of a population history]{
The relationships between populations A, B and C are depicted.
Note how branches of the same color can be simulated in parallel if there is no migration.
}
\label{fig:pop_hist}
\end{figure}

To parallelize the simulations, we need to
(i) simulate each branch of the tree independently and
(ii) put together the bits that are simulated independently at the end.
There are many ways to parallelize the simulations, but see \citet{rodrigues_vignette_2021} for an idea.
Below, I will describe the operation for putting together two (or more) tree sequences,
which I implemented in \tskit (see \url{https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.union} for the documentation).
A description of this operation will appear in an upcoming publication describing the tree sequence toolkit library (\tskit).

The information underlying a tree sequence is stored as a collection of tables which define different features.
This simple tabular format allows for rapid accessing of information.
The main components are the Node, Edge, Site and Mutation tables.
A haploid genome can be thought of as a node in a tree, which exists at a particular time.
An edge defines genetic inheritance between two nodes, and it consists of a parent node, a child node, and the left and right coordinates (along a chromosome) over which the child genome inherited from the parent genome.
A site is a location in the genome, and a mutation defines a change of state at a particular node.
See \lcref{fig:example_ts} for an example tree sequence and \lcref{fig:table_collection} for the corresponding collection of tables.
Note how information about the trees are completely independent from the notion of genetic variation (defined by the Site and Mutation tables).

\begin{figure}[htp]
\centering
\begin{subfigure}[b]{.45\linewidth}
\includegraphics[width=\linewidth]{{union_example/example_ts.pdf}}
\caption{An example tree sequence.}\label{fig:example_ts}
\end{subfigure}
\begin{subfigure}[b]{.45\linewidth}
\includegraphics[width=\linewidth]{{union_example/tables.pdf}}
\caption{The underlying collection of tables.}\label{fig:table_collection}
\end{subfigure}
\caption[An example tree sequence and its tabular encoding]{
The relationship between four samples are depicted in a sequence of trees.
Note how because of shared ancestry,
mutations that are common to many of the samples need to be represented only once.
However, in a matrix of variant sites against samples, these mutations would be repeated unnecessarily.
}
\label{fig:example_ts_and_tc}
\end{figure}


After simulating two independent populations,
one might want to obtain the node-wise union of these two tree sequences (\eg for analyzing patterns of between population variation).
This is almost as simple as concatenating the Node, Edge, Site and Mutation tables.
However, nodes from the second tree sequence need to be re-enumerated, and this new numeric order needs to be propagated onto the other tables.
I simulated the history of two populations independently using \msprime \citep{kelleher_efficient_2016, baumdicker_efficient_2022} \plcref{fig:ts1,fig:ts2}.
Then, I used the union operation to merge the two tree sequences \plcref{fig:tsu}.
Lastly, I added the pre-split history to the merged tree sequence so that all samples coalesce back-in-time \plcref{fig:tsu_recap}.
See the code in Python to reproduce this analysis \plcref{code:union}.
By doing the simulation this way, we are able to run the simulation of each extant population in parallel.
In the case of large simulations (as we will see in \lcref{chapter:greatapes}),
this is essential so that we can distribute the memory usage over different computer nodes.

\begin{lstlisting}[language=Python, caption={Python code to union two independently simulated populations using \msprime and \tskit.}, label=code:union, breaklines=true]
import msprime
import numpy as np
import tskit

# First, we build an msprime.Demography object with our two-population history that splits 100,000 units of time ago
demography = msprime.Demography()
demography.add_population(name=``A'', initial_size=10_000)
demography.add_population(name=``B'', initial_size=100)
demography.add_population(name=``C'', initial_size=100_000)
demography.add_population_split(time=100_000, derived=[``A'', ``B''], ancestral=``C'')

# Now, we can simulate the histories of the two extant populations independently
ts1 = msprime.sim_ancestry(samples={``A'':4}, ploidy=1, demography=demography, sequence_length=50, recombination_rate=1e-6, random_seed=123)
ts2 = msprime.sim_ancestry(samples={``B'':4}, ploidy=1, demography=demography, sequence_length=50, recombination_rate=1e-6, random_seed=1)

# Then, we can union these two tree sequences.
# Note that because the two populations do not share any history,
# we specify a `node_mapping' in which none of the nodes in `ts2' have an equivalent in `ts1'.
tsu = ts1.union(ts2, node_mapping=np.full(ts2.num_nodes, tskit.NULL), add_populations=False)

# Finally, we can add the pre-split history to the union'ed tree sequence with msprime.sim_ancestry.
tsu_recap = msprime.sim_ancestry(initial_state = tsu, demography=demography, random_seed=3)
\end{lstlisting}

\begin{figure}
\centering
\begin{minipage}{0.45\textwidth}%
\begin{subfigure}[b]{\linewidth}
\includegraphics[width=\linewidth]{union_example/ts2.pdf}
\caption{Tree sequence of population 1.}\label{fig:ts1}
\end{subfigure}
\end{minipage}
\begin{minipage}{0.45\textwidth}%
\begin{subfigure}[b]{\linewidth}
\includegraphics[width=\linewidth]{union_example/ts1.pdf}
\caption{Tree sequence of population 2.}\label{fig:ts2}
\end{subfigure}
\end{minipage}

\begin{subfigure}[b]{.9\textwidth}
\includegraphics[width=\linewidth]{union_example/tsu.pdf}
\caption{Union of tree sequences 1 and 2.}\label{fig:tsu}
\end{subfigure}

\begin{subfigure}[b]{0.9\textwidth}
\includegraphics[width=\linewidth]{union_example/tsu_recap.pdf}
\caption{Adding the pre-split history to the union.}\label{fig:tsu_recap}
\end{subfigure}

\caption[The union operation for tree sequences]{The union operation for tree sequences.
(a) and (b) show the tree sequences for two independently simulated populations.
(c) depicts the union of these tree sequences.
(d) shows the merged tree sequences with added history for the ancestral population.}
\label{fig:union_op}
\end{figure}

A trickier use case for the node-wise union operation is when there are two tree sequences which share part of their histories.
That is, imagine you simulate the ancestor population of A and B, and then start two new independent simulations of A and B.
In this case, it is necessary to defined which portions are not shared between the two tree sequences,
so that the histories can be joined properly.
To define the non-shared portions, it is necessary to specify which nodes in one tree sequence are equivalent to those in the other.
Then, the portions that are not shared from one tree sequence are copied over onto the other.
New nodes are given a new numeric identifier which are propagated to the other tables.
Refer to \citet{rodrigues_vignette_2021} for a concrete example on how to union tree sequences with some shared history.

\section{Towards more realistic and reproducible simulations: Introducing models of selection to Stdpopsim}

Population genetics can aid in identifying the genetic basis of adaptation.
As a new beneficial mutation increases in frequency due to positive selection,
it wipes out genetic variation surrounding the selected site \citep{smith_hitch-hiking_1974, kaplan_hitchhiking_1989}.
This genetic footprint can be used to infer past selection events in natural populations \citep{nielsen_estimation_2000, hernandez_classic_2011, garud_recent_2015, schrider_soft_2017, przeworski_signature_2002, enard_genome-wide_2014, williamson_evidence_2014}.
Our ability to detect selective sweeps depends on many parameters, such as the recombination rate, the time since fixation, and the demographic history.
The recombination rate determines the width of a selective signal (together with the strength of selection) \citep{kaplan_hitchhiking_1989}.
As time passes by, new mutations are accumulated restoring pre-sweep levels of genetic variation and erasing sweep signatures.
Demographic events can leave footprints similar to a selective sweep, confounding detection \citep{przeworski_signature_2002, jensen_distinguishing_2005}.
Conversely, sweep signatures can be erased by recent demographic events, such as bottlenecks.

Another process that can impact our ability to detect sweeps is background selection.
Background selection is the process whereby neutral genetic variation linked to deleterious mutations are lost \citep{charlesworth_effect_1993}, and the extent of this effect is also modulated by strength of selection and recombination rate.
Because of this relationship with recombination rate, background selection can confound the search for selective sweeps \citep{andolfatto_adaptive_2001}.
This process seems to be pervasive in multiple species; so much so that background selection may be considered a better null hypothesis in population genomic studies rather than neutrality \citep{comeron_background_2017}.

Background selection may be discernible from selective sweeps, specially when considering more realistic models \citep{schrider_background_2020}.
Previous studies that found background selection can confound sweep calling assumed that the region constrained by selection is flanked by large streches of neutral sites,
but the locations of exons and other constrained elements are usually scattered throughout the chromosome.
In such cases, sweeps can be separated from constraint more easily.

To better understand the role of positive selection in shaping genetic variation across the genome,
it is necessary to better delineate how different processes impact our ability to detect selective sweeps.
I present below our efforts to include models of natural selection to \stdpopsim, our community-driven library of evolutionary models.
Colleagues and I included the ability to easily simulate background selection and selective sweeps using previously published distribution of fitness effects (DFEs).
I then leveraged this new feature to understand how power to detect selective sweeps varies over a realistic looking human chromosome,
with recombination rate variation and a map of constraint taken from human annotations.
These results will appear in an upcoming publication of the \stdpopsim consortium.

\subsection{Adding selection to simulations using Stdpopsim}

There are two main ways of adding selection to a simulation in stdpopsim: (i) it is possible to specify a distribution of fitness effects (DFE) for new mutations that can occur across the genome or over some pre-specified regions; and (ii) we can introduce and track a single mutation to a population, as one would need to study selective sweeps.
Beyond the machinery to actually perform these two kinds of simulations, we now added to Stdpopsim a library of previously published distributions of fitness effects and genomic annotations (\eg exon and intron coordinates) which can be used to build realistic models with selection.
See our Tutorial for more details on how to implement models with selection in stdpopsim: \url{https://popsim-consortium.github.io/stdpopsim-docs/stable/tutorial.html#incorporating-selection}.

\subsection{Analysis of power to detect sweeps along realistic chromosomes}

To demonstrate the utility of stdpopsim, we produced maps of power to detect sweeps along a realistic chromosome.
We ran simulations of human chromosome 1 using the three population out-of-Africa model (identified as OutOfAfrica\_3G09 in the stdpopsim library; \cite{gutenkunst_inferring_2009}) and the genetic map estimated in \citet{frazer_second_2007} (HapMapII\_GRCh38).
We had three classes of simulations: (i) neutral (with the specified demography and genetic map), (ii) background selection, in which we applied the DFE estimated in \citet{kim_inference_2017} (Gamma\_K17) to exon annotations (taken from Ensembl; ensembl\_havana\_104\_exons), and (iii) sweep with background selection, in which we simulated the hard sweep as described with the addition of constraint in exonic regions (same as for \emph{ii}).
For computational efficiency, we simulated a 5Mb region flanking 100 evenly distributed points along the chromosome 1.

When simulating truncated regions with selection, it is necessary to consider a boundary effect effect:
a new deleterious mutation will be linked to fewer neutral mutations if it arises near the ends of a chromosome,
so linked selection is not as efficient.
Thus, we first established this effect in simulations in which new deleterious mutations can happen anywhere in the chromosome with the same DFE as in our main simulations, but with constant recombination rate.
We found that by adding a buffer surrounding our focal simulated regions of 2.5cM it is possible to mitigate the edge effect \plcref{fig:boundary}.
We rescaled the simulations to reduce computational cost by simulating smaller populations (factor of 2) while increasing times, mutation and recombination rates, and selection coefficients by the same amount (see \cite{uricchio_robust_2014} and \cite{adrion_community-maintained_2020} for more details).
In total, we produced 200 replicates for each class at each poisition, totalling 60,000 simulations.

\begin{figure}[htp]
\includegraphics[width=0.9\linewidth]{{sweep_figs/boundary_effect_bgs.png}}
\centering
\caption[Boundary effects in a background selection simulation]{
Linked selection is not as efficient near the ends of chromosomes.
Note how the boundary effect plateaus around 2Mb from the ends (assuming a constant recombination rate).
}
\label{fig:boundary}
\end{figure}

We computed three statistics used for finding selective sweeps along the simulated regions: (i) nucleotide diversity, (ii) the probability of a sweep using a machine learning sweep classifier called diploSHIC \citep{schrider_shic_2016, kern_diploshic_2018}, and (iii) the composite likelihood ratio (CLR) \citep{nielsen_genomic_2005}. %TODO:add citations
Nucleotide diversity was computed directly from the underlying tree sequences using \tskit \citep{ralph_efficiently_2020} over 10 equally sized and non-overlapping subwindows.
diploSHIC was applied over 15 partially overlapping subwindows.
CLR was computed at 21 uniformly distributed points along the simulated region.

We then classified the central subwindow as a sweep or not sweep.
To do so, we empirically determined the distribution of each statistic under a null model and defined the top $5\%$ quantile as a cutoff to call a sweep.
We considered two null models: neutral and background selection (as explained before).
We computed the power to detect a sweep at each one of the 100 locations along chromosome 1, which is the proportion of simulations with a sweep that were correctly classified as a sweep.

There is significant amount of variation in the power to detect sweeps along human chromosome 1 \plcref{fig:chr1power}.
Sweeps are easier to detect in YRI than in CEU, most likely because of the higher effective population size of YRI.
Variation in power varies across statistics, with nucleotide diversity yielding the greatest variation in power.
The power varies tremendously when using nucleotide diversity to call sweeps, ranging from $0\%$ to $100\%$.
diploSHIC has good power overall, reaching close to $100\%$ at many locations with YRI demography, but it falters at around $15\%$ of the locations where it dips below $75\%$.
CLR, on the other hand, maintains a good average power of around $80\%$ and it does not vary as much.

\begin{figure}[htp]
\includegraphics[width=0.9\linewidth]{{sweep_figs/chr1_power.pdf}}
\centering
\caption[Power to detect a selective sweep along chromosome 1]{
There is variation in the power to detect sweeps along human chromosome 1 for all three statistics considered (CLR, diploSHIC probability of a sweep, and nucleotide diversity).
Note that power is computed by assuming either a neutral (blue) or background selection (red) null model.
}
\label{fig:chr1power}
\end{figure}

Using background selection as a null model decreases power for both CLR and Diversity.
The effects of background selection on these two statistics are similar to a sweep,
so using this null model leads to more stringent cutoffs to call sweeps.
diploSHIC behaves differently: using background selection as a null model increases power.
This indicates that background selection does not confound sweep detection using diploSHIC,
as has been hypothesized before \citep{schrider_background_2020}.
It is encouraging that a composite statistic such as diploSHIC can discern between modes of selection,
which has been a major problem in population genetics.

To gain a better understanding at which genomic features affect variation in power,
we plotted power against recombination rates and percentage of exon overlap.
We found that sweeps placed regions of higher recombination are generally harder to detect \plcref{fig:power_rec}.
diploSHIC is not as affected by recombination rates, probably because it can distinguish between locations directly affected by sweeps and locations linked to sweeps.

\begin{figure}[htp]
\includegraphics[width=0.9\linewidth]{{sweep_figs/relationship_power_cM.pdf}}
\centering
\caption[Relationship between power and genetic recombination]{
Recombination rates affect power to detect sweeps.
The scatterplots depict the relationship between the genetic recombination rate and power for all three statistics (CLR, diploSHIC and Diversity).
Note that power is computed by assuming either a neutral (blue) or background selection (red) null model.
}
\label{fig:power_rec}
\end{figure}

This decrease in power with high recombination rates is caused by two things:
(i) the effect of recombination rate on the variance of statistics under the null, and
(ii) the effect of recombination rate on the mean of the statistics under a sweep.
With lower recombination rates, we expect the variance of statistics to increase (due to coalescent noise).
Indeed, both nucleotide diversity and diploSHIC probability of a sweep vary more in lower recombination regions \plcref{fig:stat_rec_null}.
CLR is not as affected, and this is probably why power does not vary as much along the chromosome.
In general, we expect regions of lower recombination to be more affected by a sweep,
because it is harder for linked variation to escape a sweep.
This is true for both nucleotide diversity and CLR, and we see that the median value for these statistics decrease with higher recombination rates \plcref{fig:stat_rec_sweep}.
However, the probability of sweep computed by diploSHIC increases with higher recombination rates in our sweep simulations.
Because diploSHIC models the difference between regions directly affected by a sweep and regions linked to a sweep, it is able to distinguish between classes better in regions of higher recombination rate.

\begin{figure}[htp]
\includegraphics[width=0.9\linewidth]{{sweep_figs/neutral_relationship_stat-vals_cM.pdf}}
\centering
\caption[Relationship between statistics and recombination rates under neutraility]{
Recombination rates affect the variance of statistics under neutrality.
Red lines delimit the central $90\%$ of the distribution for the three statistics at each recombination rate.
}
\label{fig:stat_rec_null}
\end{figure}

\begin{figure}[htp]
\includegraphics[width=0.9\linewidth]{{sweep_figs/sweep_relationship_stat-vals_cM.pdf}}
\centering
\caption[Relationship between statistics and recombination rates under the sweep model]{
Recombination rates affect the median of statistics under the sweep model.
The blue line is the median of the distribution for the three statistics at each recombination rate.
}
\label{fig:stat_rec_sweep}
\end{figure}

Taken together, our results demonstrate that power to detect sweeps varies considerably along realistic looking chromosomes.
Indeed, power along chromosomes may vary almost as much as they do with different demographic models.
It is necessary for future studies to consider these effects more carefully,
and we provide the tools to do so using the \stdpopsim library and software package.
It is possible that previous scans have only found sweeps where they can be found.
Thus, the effects of positive selection may have been understated in the literature,
and important sweeps might have not been correctly identified.
It seems important to better investigate whether ascertainment biases in reported sweeps correlate with power along chromosomes,
and how this varies with different methods and across species.

